staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -137,14 +137,14 @@ func ConnectResource(connecter rest.Connecter, scope *RequestScope, admit admiss
 			userInfo, _ := request.UserFrom(ctx)
 			// TODO: remove the mutating admission here as soon as we have ported all plugin that handle CONNECT
 			if mutatingAdmission, ok := admit.(admission.MutationInterface); ok {
-				err = mutatingAdmission.Admit(admission.NewAttributesRecord(opts, nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Connect, false, userInfo), scope)
+				err = mutatingAdmission.Admit(admission.NewAttributesRecord(opts, nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Connect, nil, false, userInfo), scope)
```

Comment:
I was on the fence whether the connect options struct should go in the object field or the options field (We have to keep the v1beta1 admission review object the same, but we could have changed that for v1 admissionreview objects if we wanted)

From the [KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/00xx-admission-webhooks-to-ga.md#graduation-criteria):
> There are still open questions that need to be addressed and updated in this KEP before graduation:
> 
> * ConnectOptions is sent as the main object to the webhooks today (and it is mutable). Should we change that and send parent object as the main object?

The thing that swayed me is what we publish in openapi:
```json
    "/api/v1/nodes/{name}/proxy": {
      "delete": {
        "consumes": [
          "*/*"
        ],
        "description": "connect DELETE requests to proxy of Node",
        "operationId": "connectCoreV1DeleteNodeProxy",
        "produces": [
          "*/*"
        ],
        "responses": {
          "200": {
            "description": "OK",
            "schema": {
              "type": "string"
            }
          },
          "401": {
            "description": "Unauthorized"
          }
        },
        "schemes": [
          "https"
        ],
        "tags": [
          "core_v1"
        ],
        "x-kubernetes-action": "connect",
        "x-kubernetes-group-version-kind": {
          "group": "",
          "kind": "NodeProxyOptions",
          "version": "v1"
        }
      },
```

the object we send in the admissionreview should have a GVK that matches the published GVK for the endpoint/operation

can you open a PR updating the KEP to detail the options behavior for Connect requests and removing the open question for that item?

Reply:
I've opened https://github.com/kubernetes/enhancements/pull/1062. Let me know if I misunderstood the intent.